dask/dataframe/core.py

```diff
@@ -4829,6 +5170,21 @@ def set_index(
                     f"Data has no column '{other}': use any column of {list(self.columns)}"
                 )
 
+            # Use partition statistics to check if new index is already sorted
+            if not pre_sorted and divisions is None:
+                try:
```

Comment:
This is very exciting to see.

From an API standpoint, I'd like to have a way to determine if a partition is sorted without possibly raising a `KeyError` in normal usage. That is to say, I think this is a totally reasonable request of the metadata, so it shouldn't raise a `KeyError`.